### PR TITLE
Update default generator regexp.

### PR DESCRIPTION
**Pull request by Natim**
r? @ametaireau

### CODE REVIEW

**Comment by ametaireau** [comment]
so, this isn't tested anywhere?

File: cliquet/storage/generators.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Generator(object):
     resource level to validate record id in requests paths.
     """
 
-    regexp = r'^[a-zA-Z0-9\-]+$'
+    regexp = r'^[a-zA-Z0-9][a-zA-Z0-9_-]*$'
```
Comment (leplatrem) [inline cliquet/storage/generators.py]: Are you sure there is no need for blackslash ?

...would definitely require a small test :) Especially to assert the difference between first and other chracacters

Comment (Natim) [inline cliquet/storage/generators.py]: Yes I am sure we don't need backslash because the dash is at the end of the list.
I will add a test.

**Comment by Natim** [comment]
I have added a test.

File: cliquet/tests/test_storage.py [hunk 2]
```diff
@@ -29,6 +29,17 @@ def __call__(self):
 
         self.assertRaises(AssertionError, Dumb)
 
+    def test_default_generator_allow_underscores_dash_alphabet(self):
+        class Dumb(generators.Generator):
+            def __call__(self):
+                return '1234'
```
Comment (ametaireau) [inline cliquet/tests/test_storage.py]: Not sure to understand why you actually need this. you probably can just test `generators.Generator().match()`

Comment (Natim) [inline cliquet/tests/test_storage.py]: Actually I tried this and I had a NotImplementedError.

**Comment by leplatrem** [comment]
r+

**Pull request merged**